astroid/inference.py

```diff
@@ -358,10 +356,10 @@ def infer_attribute(self, context=None):
             pass
         finally:
             context.boundnode = old_boundnode
-    return dict(node=self, context=context)
+    return InferenceErrorInfo(node=self, context=context)
 
 
-nodes.Attribute._infer = decorators.raise_if_nothing_inferred(
+nodes.Attribute._infer = decorators.raise_if_nothing_inferred(  # type: ignore[assignment]
     decorators.path_wrapper(infer_attribute)
 )
```

Comment:
Can you add a comment here (and for `nodes.Subscript._infer` that the decorator order is important?